src/plugins/intel_cpu/src/transformations/snippets/tpp/op/factory.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "factory.hpp"
+#include "eltwise.hpp"
+#include "reduce.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/type_relaxed.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace tpp {
+namespace op {
+namespace {
+struct CustomPowerStaticBuilder : public TPPNodeFactory::TPPCustomBuilder {
+    CustomPowerStaticBuilder() : TPPNodeFactory::TPPCustomBuilder() {
+        matcher = [](const std::shared_ptr<ov::Node>& n) {
+            std::set<float> supported_power{-1, 2, 0.5};
+            const auto& power_static = ov::as_type_ptr<const snippets::op::PowerStatic>(n);
+            return power_static && supported_power.count(power_static->get_power());
+        };
+        builder = [](const std::shared_ptr<ov::Node>& n) {
+            const auto& power_static = ov::as_type_ptr<snippets::op::PowerStatic>(n);
+            OPENVINO_ASSERT(power_static, "Attempt to create TPP node from unsupported input in power_static_builder");
+            const auto power = power_static->get_power();
+            const auto& input = n->get_input_source_output(0);
+            std::shared_ptr<ov::Node> tpp_node{nullptr};
+            if (power == -1.f)
+                tpp_node = std::make_shared<Reciprocal>(input);
+            else if (power == 2.f)
+                tpp_node = std::make_shared<Square>(input);
+            else if (power == 0.5f)
+                tpp_node = std::make_shared<SquareRoot>(input);
+            OPENVINO_ASSERT(tpp_node, "Failed to create TPP in power_static_builder");
+            return tpp_node;
+        };
+    }
+};
+
+} // namespace
+// TODO: it might be more convenient to use template with a non-type parameter that describes the number of arguments and to provide
+// partial specialization of the template for 1 and args. Note however that functional templates do not support partial specialization,
+// do we will need to create a dedicated class for that. Discuss if the templated solution is better on review.
+#define CREATE_UNARY_TPP_NODE(tpp_node_type) \
+    [](const std::shared_ptr<ov::Node>& ngraph_node) -> std::shared_ptr<ov::Node> { \
```

Comment:
Minor: since `ngraph` mentions are mostly deleted from the project, let's avoid such mentions in new code
```suggestion
    [](const std::shared_ptr<ov::Node>& node) -> std::shared_ptr<ov::Node> { \
```
The same for other lambda functions

Reply:
Ok, done.
BTW, do you know why the `ngraph` was removed?